devimint/src/tests.rs

```diff
@@ -590,6 +537,7 @@ pub async fn upgrade_tests(process_mgr: &ProcessManager, binary: UpgradeTest) ->
                 }))
                 .await?;
 
+                dev_fed.fed.await_gateways_registered().await?;
```

Comment:
This solved some flakiness in the gateway upgrade tests, might have been https://github.com/fedimint/fedimint/issues/6718